catalog/registry/catalog-registry-identificationplugin/src/main/java/ddf/catalog/registry/identification/IdentificationPlugin.java

```diff
@@ -0,0 +1,183 @@
+import javax.xml.bind.JAXBElement;
+
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import com.google.common.io.FileBackedOutputStream;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.registry.common.RegistryConstants;
+import ddf.catalog.registry.common.metacard.RegistryMetacardImpl;
+import ddf.catalog.registry.common.metacard.RegistryObjectMetacardType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExternalIdentifierType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+
+public class IdentificationPlugin implements PreIngestPlugin {
+
+    private Parser parser;
+
+    private ParserConfigurator configurator;
+
+    @Override
+    public CreateRequest process(CreateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Metacard> metacards = input.getMetacards();
+        Map<String, Serializable> properties = input.getProperties();
+        List<Metacard> updatedMetacards = new ArrayList<>();
+
+        for (Metacard metacard : metacards) {
+            Set<String> metacardTags = metacard.getTags();
+            if (metacardTags.contains(RegistryConstants.REGISTRY_TAG)) {
+                updatedMetacards.add(setMetacardExtID((RegistryMetacardImpl) metacard));
+            } else {
+                updatedMetacards.add(metacard);
+            }
+
+        }
+        CreateRequest updatedRequest = new CreateRequestImpl(updatedMetacards, properties);
+
+        return updatedRequest;
+    }
+
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteRequest process(DeleteRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        return input;
+    }
+
+    private Metacard setMetacardExtID(RegistryMetacardImpl metacard) throws StopProcessingException {
+
+        String metacardID = metacard.getId();
+        String metadata = metacard.getMetadata();
+        String registryID = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+
+        InputStream inputStream = new ByteArrayInputStream(metadata.getBytes(Charsets.UTF_8));
+        FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000);
+
+        try {
+            JAXBElement<RegistryObjectType> registryObjectTypeJAXBElement = parser.unmarshal(
```

Comment:
I'm on my phone so may be wrong, but isn't this Parser set through blueprint injection? So for it to be null you'd have blown up initializing this?